### PR TITLE
Improve yaml-cpp config search and CMake messaging

### PR DESCRIPTION
**Pull request by michdolan**
- Fixes issue where yaml-cpp will be detected from a CMake config in a previous OCIO build `ext/dist` dir, which was caused by the config search checking the CMake User Package Registry.
- Cleans up Python package checking messaging. Previously appeared to be an error, and now the message is more concise and aligned with other package check messages since the traceback is now omitted.
- Make OCIO_INSTALL_EXT_PACKAGES case insensitive, and error for an invalid value

Signed-off-by: Michael Dolan <michdolan@gmail.com>

### CODE REVIEW

**Comment by hodoulp** [comment]
Good catch.

Here is the link to understand the cmake feature: https://cmake.org/cmake/help/v3.12/manual/cmake-packages.7.html#user-package-registry

@michdolan Is there something to do for the 'system package registry' ?

https://cmake.org/cmake/help/v3.12/command/find_package.html
https://cmake.org/cmake/help/v3.12/manual/cmake-packages.7.html#system-package-registry

File: CMakeLists.txt [hunk 1]
```diff
@@ -206,8 +206,24 @@ endif(NOT HAVE_SSE2)
...
+# Support options
+set(_EXT_OPTIONS NONE MISSING ALL)
+string(REPLACE ";" ", " _EXT_OPTIONS_STR "${_EXT_OPTIONS}")
+
+# Default option
+set(_OCIO_INSTALL_EXT_PACKAGES "MISSING")
+
+if(DEFINED OCIO_INSTALL_EXT_PACKAGES)
+    # Case insensitive match
+    string(TOUPPER "${OCIO_INSTALL_EXT_PACKAGES}" _OCIO_INSTALL_EXT_PACKAGES)
```
Comment (KelSolaar) [inline CMakeLists.txt:218]: Nice! 👍

**Comment by michdolan** [comment]
It seems like using the system registry is more explicit since CMake provides no interface to do it. The build setup would have to manually add a registry entry on Windows for example. For now, I think we could leave it alone, but it is good to keep it in mind for the future.

File: CMakeLists.txt (hunk 1 above)
Comment (michdolan) [reply CMakeLists.txt:218]: Was great timing for your feedback :)

Comment (KelSolaar) [reply CMakeLists.txt:218]: Yep!

Comment (hodoulp) [reply CMakeLists.txt:218]: Change fixing #1103

**Comment by hodoulp** [comment]
Great. Monday, I will check all of that on macOS & Windows machines.

**Comment by michdolan** [comment]
Good find @hodoulp . I think that sounds reasonable.

**Comment by michdolan** [comment]
Swapped to disabling both package registries globally, which also resolves the issue in my tests.

Note: I noticed that in more recent CMake versions, the user package registry variable has been deprecated in favor of similar variables which doesn't contain `_NO_` and needs to be set to `OFF` instead of `ON`:

https://cmake.org/cmake/help/v3.18/manual/cmake-packages.7.html#disabling-the-package-registry

**Pull request merged**